File: Searching/binary_search.java
package Searching;

public class binary_search {
    public static int search(int [] arr,int value)
    {
        int start=0;
        int end=arr.length-1;
        int middle=(start+end)/2;
        while(arr[middle]!=value && start<=end){
            if(arr[middle]>value)
            {
                end=middle-1;
            }
            else
            {
                start=middle+1;
            }
            middle=(start+end)/2;
        }
        if(arr[middle]==value)
        {
            System.out.println("Element found");
            return middle;
        }
        else
        {
            System.out.println("Element not found!");
            return -1;
        }
    }
    public static void main(String args[])
    {
        int [] arr={1,9,2,8};
        System.out.println(search(arr,9));
        System.out.println(search(arr,7));
    }
}

File: 9_Searching/binary_search.py
def binary_search(arr,value):
    s=0
    e=len(arr)-1
    mid=(s+e)//2
    while s<=e and arr[mid]!=value:
        if arr[mid]>value:
            e=mid-1
        else:
            s=mid+1
        mid=(s+e)//2
    
    if arr[mid]==value:
        return "found"
    else:
        return "not found"

print(binary_search([9,18,27],3))
print(binary_search([9,18,27],9))
